week-3/todo-list/script.js

```diff
@@ -1,25 +1,126 @@
-function populateTodoList(todos) {
-  let list = document.getElementById("todo-list");
-  // Write your code to create todo list elements with completed and delete buttons here, all todos should display inside the "todo-list" element.
-}
+document.addEventListener("DOMContentLoaded", function () {
+  const todoForm = document.getElementById("todo-form");
+  const todoInput = document.getElementById("todoInput");
+  const deadlineInput = document.getElementById("deadlineInput");
+  const todoList = document.getElementById("todo-list");
+  const removeAllCompletedButton = document.getElementById(
+    "remove-all-completed"
+  );
+  const addTodoBtn = document.getElementById("addTodoBtn");
 
-// These are the same todos that currently display in the HTML
-// You will want to remove the ones in the current HTML after you have created them using JavaScript
-let todos = [
-  { task: "Wash the dishes", completed: false },
-  { task: "Do the shopping", completed: false },
-];
+  let todos = [
+    // { task: "Wash the dishes", completed: false, deadline: "" },
+    // { task: "Do the shopping", completed: false, deadline: "" },
+  ];
 
-populateTodoList(todos);
+  function populateTodoList() {
+    todoList.innerHTML = "";
 
-// This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
-function addNewTodo(event) {
-  // The code below prevents the page from refreshing when we click the 'Add Todo' button.
-  event.preventDefault();
-  // Write your code here... and remember to reset the input field to be blank after creating a todo!
-}
+    todos.forEach((todo, index) => {
+      const li = document.createElement("li");
+      const checkbox = createCustomCheckbox();
+      const label = document.createElement("label");
+      const deadlineText = document.createElement("span");
+      const deleteIcon = document.createElement("span");
+
+      checkbox.type = "checkbox";
+      checkbox.checked = todo.completed;
+      checkbox.addEventListener("change", () => toggleTodoCompleted(index));
+
+      label.textContent = todo.task;
+      label.style.textDecoration = todo.completed ? "line-through" : "none";
+
+      // Display deadline if set
+      if (todo.deadline) {
+        deadlineText.textContent = ` | ${formatDeadline(
+          new Date(todo.deadline)
+        )}`;
+      }
+
+      deleteIcon.innerHTML = "&#x1F5D1;";
+      deleteIcon.style.cursor = "pointer";
+      deleteIcon.addEventListener("click", () => {
```

Comment:
Can I only delete completed todo items? What if I added a todo item by mistake and wanted to delete it, how could I achieve that?